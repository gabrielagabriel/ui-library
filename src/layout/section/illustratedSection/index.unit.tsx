import React from 'react'
import renderer from 'react-test-renderer'

import IllustratedSection from './index'

const defaultProps = {
  illustrationUrl: 'http://blablacar.com/pic.jpg',
}

describe('IllustratedSection', () => {
  it('should render default IllustratedSection section', () => {
    const section = <IllustratedSection {...defaultProps}>Hello World</IllustratedSection>
    const renderedSection = renderer.create(section).toJSON()
    expect(renderedSection).toMatchSnapshot()
  })

  it('should render an IllustratedSection section with an Avatar', () => {
    const section = (
      <IllustratedSection {...defaultProps} isAvatar>
        Hello World
      </IllustratedSection>
    )
    const renderedSection = renderer.create(section).toJSON()
    expect(renderedSection).toMatchSnapshot()
  })
})
