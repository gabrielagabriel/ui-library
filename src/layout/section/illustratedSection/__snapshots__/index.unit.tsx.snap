// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IllustratedSection should render an IllustratedSection section with an Avatar 1`] = `
.c1 .section-content {
  padding-left: 24px;
  padding-right: 24px;
}

.c2 {
  border: 2px dashed #DDD;
  box-sizing: border-box;
  border-radius: 50%;
  position: relative;
  width: 48px;
  height: 48px;
}

.c2.kirk-avatar--small {
  height: 40px;
  width: 40px;
}

.c2.kirk-avatar--medium {
  height: 96px;
  width: 96px;
}

.c2.kirk-avatar--large {
  height: 156px;
  width: 156px;
}

.c2.kirk-avatar--image {
  border: none;
}

.c2.kirk-avatar--image img {
  border-radius: 50%;
  width: 100%;
  height: 100%;
  object-fit: cover;
}

.c2 .kirk-badge.kirk-avatar-badge--unreadNotifications,
.c2 .kirk-badge.kirk-avatar-badge--idCheck {
  z-index: 2;
  position: absolute;
}

.c2 .kirk-badge.kirk-avatar-badge--unreadNotifications {
  top: -6px;
  right: -6px;
}

.c2 .kirk-badge.kirk-avatar-badge--idCheck {
  bottom: -3px;
  right: -3px;
  width: 18px;
  height: 18px;
  padding: 0;
  border-radius: 50%;
  background-color: #5DD167;
}

.c2.kirk-avatar--medium .kirk-badge.kirk-avatar-badge--idCheck,
.c2.kirk-avatar--large .kirk-badge.kirk-avatar-badge--idCheck {
  bottom: 0;
  right: 0;
}

.c2 .kirk-avatar--medium .kirk-badge.kirk-avatar-badge--idCheck {
  width: 24px;
  height: 24px;
}

.c2 .kirk-avatar--large .kirk-badge.kirk-avatar-badge--idCheck {
  width: 36px;
  height: 36px;
}

.c2 .kirk-avatar--medium .kirk-badge.kirk-avatar-badge--unreadNotifications {
  height: 24px;
  line-height: 23px;
  font-size: 14px;
  padding: 0 6px;
}

.c2 .kirk-avatar--large .kirk-badge.kirk-avatar-badge--unreadNotifications {
  height: 36px;
  line-height: 34px;
  font-size: 18px;
  padding: 0 8px;
}

.c0 .section-content {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0 .kirk-illustratedSection-illustration {
  margin-left: -24px;
  margin-right: -24px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c0 .kirk-illustratedSection-illustration img {
  max-width: 100%;
}

@media (min-width:800px) {
  .c1 .section-content {
    margin-left: auto;
    margin-right: auto;
    max-width: 662px;
  }

  .c1 .section-content.section-content--large {
    max-width: 1016px;
  }
}

@media (min-width:800px) {
  .c0 .section-content {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }

  .c0 .kirk-illustratedSection-illustration {
    margin-left: 0;
    margin-right: 0;
    width: 33.33%;
    padding-top: 24px;
  }

  .c0 .kirk-illustratedSection-content {
    margin-left: 24px;
    width: 66.66%;
  }
}

<div
  className="c0 c1"
  role="presentation"
>
  <article
    className="section-content section-content--large"
  >
    <div
      className="kirk-illustratedSection-illustration"
    >
      <div
        className="kirk-avatar--large kirk-avatar--image c2 kirk-avatar"
      >
        <img
          alt=""
          src="http://blablacar.com/pic.jpg"
        />
      </div>
    </div>
    <div
      className="kirk-illustratedSection-content"
    >
      Hello World
    </div>
  </article>
</div>
`;

exports[`IllustratedSection should render default IllustratedSection section 1`] = `
.c1 .section-content {
  padding-left: 24px;
  padding-right: 24px;
}

.c0 .section-content {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0 .kirk-illustratedSection-illustration {
  margin-left: -24px;
  margin-right: -24px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c0 .kirk-illustratedSection-illustration img {
  max-width: 100%;
}

@media (min-width:800px) {
  .c1 .section-content {
    margin-left: auto;
    margin-right: auto;
    max-width: 662px;
  }

  .c1 .section-content.section-content--large {
    max-width: 1016px;
  }
}

@media (min-width:800px) {
  .c0 .section-content {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }

  .c0 .kirk-illustratedSection-illustration {
    margin-left: 0;
    margin-right: 0;
    width: 33.33%;
    padding-top: 24px;
  }

  .c0 .kirk-illustratedSection-content {
    margin-left: 24px;
    width: 66.66%;
  }
}

<div
  className="c0 c1"
  role="presentation"
>
  <article
    className="section-content section-content--large"
  >
    <div
      className="kirk-illustratedSection-illustration"
    >
      <img
        alt=""
        src="http://blablacar.com/pic.jpg"
      />
    </div>
    <div
      className="kirk-illustratedSection-content"
    >
      Hello World
    </div>
  </article>
</div>
`;
